### PR TITLE
build: Migrate to devbox for tooling configurations

### PR DESCRIPTION
**Pull request by jimmidyson**
**What this PR does / why we need it**:
This PR switches to using [devbox](https://www.jetpack.io/devbox) to provide configure developer tooling, which in my experience results in a more consistent environment between dev and test machines.

This also updates the release process to use [goreleaser](https://goreleaser.com/).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**How Has This Been Tested?**:

I have tested the builds and release process in my own fork and it all worked smoothly, although I haven't been able to launch e2e tests so that could be broken - any help testing those would be appreciated 🙏 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/developer_workflow.md [hunk 1]
```diff
@@ -8,36 +8,26 @@ cd cluster-api-provider-nutanix
...
-## Prepare local clusterctl
-<pre>
-make prepare-local-clusterctl
+make build-snapshot
 </pre>
 
 ## Deploy cluster-api-provider-nutanix CRDs on test management cluster
 <pre>
-make deploy
+make dev.run-on-kind
```
Comment (dkoshkin) [inline docs/developer_workflow.md:16]: Does devbox have `gojq`?
```
make[1]: Leaving directory '/home/dkoshkin/src/github.com/nutanix-cloud-native/cluster-api-provider-nutanix'
bash: line 1: gojq: command not found
ERROR: image: "ko.local/cluster-api-provider-nutanix:" not present locally
make: *** [make/dev.mk:19: dev.run-on-kind] Error 1
```

File: docs/developer_workflow.md [hunk 2]
```diff
@@ -8,36 +8,26 @@ cd cluster-api-provider-nutanix
...
 </pre>
 
 ## Deploy test workload cluster
 Note: Update ./clusterctl.yaml with appropriate configuration before running following commands
 <pre>
-make test-clusterctl-create
+make test.clusterctl-create
```
Comment (dkoshkin) [inline docs/developer_workflow.md:22]: Missing KIND kubeconfig
```
Error: management cluster not available. Cannot auto-discover target namespace. Please specify a target namespace: invalid kubeconfig file; clusterctl requires a valid kubeconfig file to connect to the management cluster: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
```

File: docs/developer_workflow.md [hunk 3]
```diff
@@ -8,36 +8,26 @@ cd cluster-api-provider-nutanix
...
-
-## Create test management cluster
-<pre>
-make kind-create
-</pre>
-
-## Prepare local clusterctl
-<pre>
-make prepare-local-clusterctl
+make build-snapshot
```
Comment (dkoshkin) [inline docs/developer_workflow.md:11]: I don't think we even need the build anymore, the `dev.run-on-kind` target already runs `release-snapshot`

**Review by dkoshkin — COMMENTED** [review]
Great changes!

File: docs/developer_workflow.md (hunk 1 above)
Comment (jimmidyson) [reply docs/developer_workflow.md:16]: Oops! Try again, I missed it from the configuration.

**Comment by jimmidyson** [comment]
Closing in deference to #358.

**Pull request closed without merging**